src/components/textarea/textarea.tsx

```diff
@@ -1,22 +1,70 @@
-import React, { forwardRef, InputHTMLAttributes, LegacyRef } from 'react'
-import { ElTextArea, elTextAreaHasError } from './__styles__'
 import { cx } from '@linaria/core'
+import { ElTextArea, elTextAreaHasError, type TextAreaCSSProperties } from './styles'
+import { forwardRef } from 'react'
 
-export interface TextAreaProps extends InputHTMLAttributes<HTMLTextAreaElement> {
+import type { CSSProperties, TextareaHTMLAttributes } from 'react'
+
+// NOTE: We omit the `cols` prop because our text area should always grow to the width of its container.
+interface RestrictedTextareaHTMLAttributes extends Omit<TextareaHTMLAttributes<HTMLTextAreaElement>, 'cols'> {}
+
```

Comment:
**note:** I've updated how the component's prop interface is defined for two reasons:

1. To add the new `minRows` and `maxRows` props; and,
2. To use the `TextareaHTMLAttributes` instead of `InputHTMLAttributes` (the latter one is for `<input>` elements).